### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by mohamedSabry0**
In this PR:
- Classroom, Book, and Rental classes were added.
- Has-many/belongs-to relationship between Classroom and Student was created.
- Many-to-many (also has-many-through) relationship between Person and Book was created using the intermediate class Rental.

### CODE REVIEW

File: classes/book.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (mikethreels) [inline classes/book.rb]: - [ ] @rentals should have a write attribute as well. In order for the associations to work

File: classes/book.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals << rental
+    rental.book = self
+  end
```
Comment (mikethreels) [inline classes/book.rb]: - [ ] the has many part of this relationship should look a little different, please have a look at[ this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) to see what is should look like

File: classes/rental.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  attr_reader :book, :person
+  attr_accessor :date
```
Comment (mikethreels) [inline classes/rental.rb:3]: - [ ] the date instance variable should have a write attribute as well for this relationship to work correctly.

File: classes/person.rb [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 class Person < Nameable
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (mikethreels) [inline classes/person.rb:6]: - [ ] in order to make the association work there should a @rental instance variable and an add_rental method, please have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) for more details.

File: classes/person.rb [hunk 5]
```diff
@@ -2,7 +2,7 @@
...
```
Comment (mikethreels) [inline classes/person.rb]: - [ ] the rental instance variable should have a write attribute as well in order for this relationship to work.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Mohamed, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Good job starting to setup the associations.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mohamedSabry0** [comment]
Thank you for the review!

I checked the article once more, and here are my observations:
- The middle class (in this case Rental) has the methods needed for the associations to work.
- `add_rental` method seems to be on the "has-many" side, but only in a has-many/belongs-to relationship.
- Only `attr_reader`s are used for the instance variable @rentals.

Thus, it might work without these changes. Nevertheless, I'm going to update with the changes that were needed.
I would like more elaboration about these points if possible so that I have a better understanding of the associations.

I would like also to test if it would work with only these two changes:
- Removing `add_rental` from the book class.
- Adding the instance variable @rentals to the person class.

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾✔🎉**
Hello @mohamedSabry0 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- Descriptive PR.
- Good commit messages.
- Correct git-flow.

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**